PATCH-P0-HACK-UPSTREAM: Make federated plugin to compile

Really ugly way to force federated plugin to compile if it is choosed to be
dynamically loadable.

Maintainer: Michal Hrusecky <Michal.Hrusecky@opensuse.org>

--- storage/federated/ha_federated.cc
+++ storage/federated/ha_federated.cc
@@ -385,6 +385,12 @@
 
 #include <mysql/plugin.h>
 
+
+// Quick and ugly plugin workaround
+#ifdef MYSQL_DYNAMIC_PLUGIN
+#include "../../mysys/string.c"
+#endif
+
 /* Variables for federated share methods */
 static HASH federated_open_tables;              // To track open tables
 pthread_mutex_t federated_mutex;                // To init the hash
